File: goblin-queue/src/main/java/org/goblinframework/queue/api/GoblinQueueProducer.java
package org.goblinframework.queue.api;

import org.goblinframework.queue.GoblinMessageSerializer;
import org.goblinframework.queue.QueueSystem;

import java.lang.annotation.*;

@Documented
@Target({ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(GoblinQueueProducers.class)
public @interface GoblinQueueProducer {

  QueueSystem system() default QueueSystem.KFK;

  String config() default "primary";

  String queue();

  GoblinMessageSerializer serializer() default GoblinMessageSerializer.HESSIAN2;

  boolean enabled() default true;
}

File: goblinframework-queue/src/main/java/org/goblinframework/queue/kafka/producer/GoblinQueueProducer.kt
package org.goblinframework.queue.kafka.producer

import org.goblinframework.queue.kafka.QueueSystem

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@Repeatable
annotation class GoblinQueueProducer (

    val queue: String,             // queue name

    val queueSystem: QueueSystem = QueueSystem.RMQ,  // queue system type

    val name: String,

    val enable: Boolean = true
)